easybuild/easyconfigs/s/Safetensors/Safetensors-0.4.3-foss-2023a-CUDA-12.1.1.eb

```diff
@@ -0,0 +1,301 @@
+easyblock = "CargoPythonBundle"
+
+name = 'Safetensors'
+version = '0.4.3'
+versionsuffix = '-CUDA-%(cudaver)s'
+_rustver = '1.75.0'
+
+homepage = 'https://huggingface.co/docs/safetensors'
+description = """Safetensors is a new simple format for storing tensors safely (as opposed to
+pickle) and that is still fast (zero-copy). Safetensors is really fast.
+
+This variant of Safetensors is installed with support for numpy and PyTorch
+"""
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+
+builddependencies = [
+    ('Rust', _rustver),
+    ('maturin', '1.4.0', '-Rust-' + _rustver),
+]
+
+dependencies = [
+    ('CUDA', '12.1.1', '', SYSTEM),
+    ('Python', '3.11.3'),
+    ('SciPy-bundle', '2023.07'),
+    ('PyTorch', '2.1.2', versionsuffix),  # optional
+    # ('jax', 'xxx', versionsuffix),  # optional, add 'jax' to 'use_pip_extras'
+    # ('TensorFlow', 'xxx', versionsuffix),  # optional, add 'tensorflow' to 'use_pip_extras'
```

Comment:
```suggestion
```

I think these can be opt-in at runtime, which would safe us from having to make multiple versions of this (and considering the combinatorics of 3 optional things, with cuda and non-cuda, would be insane, n=2x2x2x2=ugh) 

I skimmed the source code, and it's just different plain python code wrappers for save_file and such in safetensors, and I think they are all installed unconditionally and can be combined with any (compatible) torch/numpy/tensorflow/paddle/etc.etc.

(I could be wrong, didn't test this theory yet)